clients/nutclient.cpp

```diff
@@ -1705,6 +1711,9 @@ int nutclient_get_device_num_logins(NUTCLIENT_t client, const char* dev)
 	return -1;
 }
 
+/* FIXME: Protocol update needed to handle master/manager alias
```

Comment:
Agreed. Some pragma warning for compile time will be good too

Reply:
It seems that compiler support for the feature varies: a pragma may be a hammer too blunt, impacting all users of the file (packagers, home users - non-devs just building NUT and with little say over actual use or not of the deprecated routines).

Recent compiler versions support (sometimes even in the same manner) ways of providing attributes to functions - so our and third-party consumer code that uses them would see the warning and be in position to update:

* clang and gcc allow for `__attribute__((deprecated))` decorations, with optional message:
````
void explode(void) __attribute__((deprecated("extremely unsafe, use 'combust' instead!!!")));
````
** https://clang.llvm.org/docs/LanguageExtensions.html#messages-on-deprecated-and-unavailable-attributes
** https://gcc.gnu.org/onlinedocs/gcc-4.7.0/gcc/Function-Attributes.html

* clang and MSVS allow for `__declspec(deprecated)`, not sure yet if MSVS also handles `__attribite__` in any version
````
__declspec(deprecated("Please use F2")) void F1();
````

* C++14 (for nutclient.cpp/.hpp in particular) introduces the `deprecated` keyword into C++ language

Thanks for posters to https://stackoverflow.com/questions/26429243/difference-between-pragmadeprecated-and-declspecdeprecated for providing many hints for the above summary.

So following up with this would be an interesting topic, probably with a fair bit of M4 to check which keyword best applies to the current compiler (and language revision) and peppering code with that. Or even "worse" -- at least in some cases we are talking about public headers for third-party consumers, so the complete solution should better be embedded in them.